nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ListenTCPRecord.java

```diff
@@ -0,0 +1,432 @@
+        "documents and then send another array on the same connection, as the reader would be in a bad state at that point. Records " +
+        "will be read from the connection in blocking mode, and will timeout according to the Read Timeout specified in the processor. " +
+        "If the read times out, or if any other error is encountered when reading, the connection will be closed, and any records " +
+        "read up to that point will be handled according to the configured Read Error Strategy (Discard or Transfer). In cases where " +
+        "clients are keeping a connection open, the concurrent tasks for the processor should be adjusted to match the Max Number of " +
+        "TCP Connections allowed, so that there is a task processing each connection.")
+@WritesAttributes({
+        @WritesAttribute(attribute="tcp.sender", description="The host that sent the data."),
+        @WritesAttribute(attribute="tcp.port", description="The port that the processor accepted the connection on."),
+        @WritesAttribute(attribute="record.count", description="The number of records written to the flow file."),
+        @WritesAttribute(attribute="mime.type", description="The mime-type of the writer used to write the records to the flow file.")
+})
+public class ListenTCPRecord extends AbstractProcessor {
+
+    static final PropertyDescriptor PORT = new PropertyDescriptor
+            .Builder().name("Port")
+            .description("The port to listen on for communication.")
+            .required(true)
+            .addValidator(StandardValidators.PORT_VALIDATOR)
+            .expressionLanguageSupported(true)
+            .build();
+
+    static final PropertyDescriptor READ_TIMEOUT = new PropertyDescriptor.Builder()
+            .name("Read Timeout")
+            .description("The amount of time to wait before timing out when reading from a connection.")
+            .addValidator(StandardValidators.TIME_PERIOD_VALIDATOR)
+            .defaultValue("30 seconds")
+            .required(true)
+            .build();
+
+    static final PropertyDescriptor MAX_SOCKET_BUFFER_SIZE = new PropertyDescriptor.Builder()
+            .name("Max Size of Socket Buffer")
+            .description("The maximum size of the socket buffer that should be used. This is a suggestion to the Operating System " +
+                    "to indicate how big the socket buffer should be. If this value is set too low, the buffer may fill up before " +
+                    "the data can be read, and incoming data will be dropped.")
+            .addValidator(StandardValidators.DATA_SIZE_VALIDATOR)
+            .defaultValue("1 MB")
+            .required(true)
+            .build();
+
+    static final PropertyDescriptor MAX_CONNECTIONS = new PropertyDescriptor.Builder()
+            .name("Max Number of TCP Connections")
+            .description("The maximum number of concurrent TCP connections to accept.")
```

Comment:
Should we add in the property description the comment you added in the capability description?
"In cases where clients are keeping a connection open, the concurrent tasks for the processor should be adjusted to match the Max Number of TCP Connections allowed, so that there is a task processing each connection."